sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala

```diff
@@ -309,6 +310,15 @@ class CatalystTypeConvertersSuite extends SparkFunSuite with SQLHelper {
     assert(errMsg.contains("integer overflow"))
   }
 
+  test("SPARK-35769: Truncate java.time.Period by fields of year-month interval type") {
+    Seq(YearMonthIntervalType(YEAR, YEAR) -> 12,
+      YearMonthIntervalType(YEAR, MONTH) -> 13,
+      YearMonthIntervalType(MONTH, MONTH) -> 1)
```

Comment:
The result of the conversion `Period.of(1, 1, 0)` to `INTERVAL MONTH` must be `INTERVAL '13' MONTH`. cc @cloud-fan @srielau

Reply:
Like current?